File: src/src_core/fmg/core/img/AnimatedInnerModel.cs
using System;
using System.ComponentModel;
using Fmg.Common.Geom;
using Fmg.Common.Notifier;

namespace Fmg.Core.Img {

    /// <summary> MVC: model. Animated image characteristics </summary>
    public sealed class AnimatedInnerModel : IAnimatedModel {

        /// <summary> Image is animated? </summary>
        private bool _animated;
        /// <summary> Overall animation period (in milliseconds) </summary>
        private long _animatePeriod = 3000;
        /// <summary> Total frames of the animated period </summary>
        private int _totalFrames = 30;
        private int _currentFrame = 0;

        public event PropertyChangedEventHandler PropertyChanged {
            add    { _notifier.PropertyChanged += value;  }
            remove { _notifier.PropertyChanged -= value;  }
        }
        private readonly NotifyPropertyChanged _notifier;

        #region: begin unusable code
        public SizeDouble Size {
            get { throw new NotImplementedException(); }
            set { throw new NotImplementedException(); }
        }
        public BoundDouble Padding {
            get { throw new NotImplementedException(); }
            set { throw new NotImplementedException(); }
        }
        #endregion: end unusable code

        public AnimatedInnerModel() {
            _notifier = new NotifyPropertyChanged(this);
        }

        public bool Animated {
            get { return _animated; }
            set { _notifier.SetProperty(ref _animated, value); }
        }

        /// <summary> Overall animation period (in milliseconds) </summary>
        public long AnimatePeriod {
            get { return _animatePeriod; }
            set { _notifier.SetProperty(ref _animatePeriod, value); }
        }

        /// <summary> Total frames of the animated period </summary>
        public int TotalFrames {
            get { return _totalFrames; }
            set {
                if (_notifier.SetProperty(ref _totalFrames, value))
                    CurrentFrame = 0;
            }
        }

        public int CurrentFrame {
            get { return _currentFrame; }
            set { _notifier.SetProperty(ref _currentFrame, value); }
        }

        public void Dispose() {
            _notifier.Dispose();
            GC.SuppressFinalize(this);
        }

    }

}

File: FastMines_core/src/fmg/core/img/AnimatedInnerModel.java
package fmg.core.img;

import java.beans.PropertyChangeListener;

import fmg.common.geom.BoundDouble;
import fmg.common.geom.SizeDouble;
import fmg.common.notifier.NotifyPropertyChanged;
import fmg.core.types.Property;

/** MVC: inner model. Animated image characteristics. */
final class AnimatedInnerModel implements IAnimatedModel {

    /** Image is animated? */
    @Property(PROPERTY_ANIMATED)
    private boolean animated;

    /** Overall animation period (in milliseconds) */
    @Property(PROPERTY_ANIMATE_PERIOD)
    private long animatePeriod = 3000;

    /** Total frames of the animated period */
    @Property(PROPERTY_TOTAL_FRAMES)
    private int totalFrames = 30;

    @Property(PROPERTY_CURRENT_FRAME)
    private int currentFrame = 0;

    private NotifyPropertyChanged notifier = new NotifyPropertyChanged(this);

    // #region: begin unusable code
    @Override
    public SizeDouble getSize()                 { throw new UnsupportedOperationException(); }
    @Override
    public void setSize(SizeDouble value)       { throw new UnsupportedOperationException(); }

    @Override
    public BoundDouble getPadding()             { throw new UnsupportedOperationException(); }
    @Override
    public void setPadding(BoundDouble padding) { throw new UnsupportedOperationException(); }
    // #region: end unusable code

    /** Image is animated? */
    @Override
    public boolean isAnimated() { return animated; }
    @Override
    public void setAnimated(boolean value) {
        notifier.setProperty(animated, value, PROPERTY_ANIMATED);
    }

    /** Overall animation period (in milliseconds) */
    @Override
    public long getAnimatePeriod() { return animatePeriod; }
    /** Overall animation period (in milliseconds) */
    @Override
    public void setAnimatePeriod(long value) {
        notifier.setProperty(animatePeriod, value, PROPERTY_ANIMATE_PERIOD);
    }

    /** Total frames of the animated period */
    @Override
    public int getTotalFrames() { return totalFrames; }
    @Override
    public void setTotalFrames(int value) {
        if (notifier.setProperty(totalFrames, value, PROPERTY_TOTAL_FRAMES))
            setCurrentFrame(0);
    }

    @Override
    public int getCurrentFrame() { return currentFrame; }
    @Override
    public void setCurrentFrame(int value) {
        notifier.setProperty(currentFrame, value, PROPERTY_CURRENT_FRAME);
    }

    @Override
    public void close() {
        notifier.close();
    }

    @Override
    public void addListener(PropertyChangeListener listener) {
        notifier.addListener(listener);
    }
    @Override
    public void removeListener(PropertyChangeListener listener) {
        notifier.removeListener(listener);
    }

}
